prow/plugins/config.go

```diff
@@ -60,6 +60,7 @@ type Configuration struct {
 	Cat                        Cat                    `json:"cat,omitempty"`
 	CherryPickUnapproved       CherryPickUnapproved   `json:"cherry_pick_unapproved,omitempty"`
 	ConfigUpdater              ConfigUpdater          `json:"config_updater,omitempty"`
+	Dco                        []Dco                  `json:"dco,omitempty"`
```

Comment:
I've tried to get around this, but I'm not sure how this should look like and where exactly it's implemented for `branch_protection`. Can you give me some hints or examples where can I find that or how that should look like? I've went through #9757 but it's mostly about implementing a library that does this and there's still some ongoing discussion.

Reply:
Example is here: https://github.com/kubernetes/test-infra/blob/master/prow/config/branch_protection.go

Reply:
Basic idea is having a hierarchy of embedded structs

Reply:
Look here for a sample: https://github.com/kubernetes/test-infra/blob/0e68a1e45538e8066910c92552df926418df0020/prow/config/config.go#L219-L233

Can also replace the logic in the plugins help, IMHO much easier to understand than the current impementation.

Reply:
(Use a `map[string]*DCO`)

Reply:
I've followed the example Alvaro linked to try to implement hierarchies and overrides. The way Plank does it is a little bit different compared to `branch_protection` but on the other side it seems more simple. If I implemented it correctly, I believe that we will get the effect we want and that it's going to be easier to manage DCO settings per org and repo.

The way it works is similar to Plank. The Dco field is now `map[string]*Dco` and the key scheme is:
* `*` is the default DCO configuration (don't skip DCO checks),
* if `org/repo` key exists, return it
* if `org` key exists, return it
* otherwise return default (`*`)

A potential problem here is to make this work, `Dco` has to be `map[string]*Dco`. Earlier, there have been some concerns regarding using pointers, but I think using pointers is the only way to make it work. If there is a better way, please let me know.

@alvaroaleman @stevekuznetsov PTAL and let me know what do you think and is there anything I can improve.